app/assets/javascripts/float-kit/addon/components/d-toasts.gjs

```diff
@@ -27,6 +29,9 @@ class AutoCloseToast extends Modifier {
     this.element = element;
     this.close = close;
     this.duration = duration;
+    this.progressBar = element.querySelector(
```

Comment:
We should always avoid doing this, query selector is for vanilla js, but we don't want this in ember.

I have another question before, how does it work with multiple toasts?

Reply:
Multiple toasts work fine, they will stack as normal and hover/mouseover only affects the specific toast that you are interacting with.

Reply:
Yes it's tricky... what you can do:

- `<div class="progress-bar" {{didInsert @onRegisterProgressBar}}>...</div>`
- `<toast.options.component @data={{toast.options.data}} @close={{toast.close}} @onRegisterProgressBar={{this.registerProgressBar}} />`
- move all the logic of autoclose and the template of <output>...</output> inside a <DToast /> component, that will allow use to have an instance per toast
- in `<DToast />` have this action: `registerProgressBar(element) { this.progressBar =  element; }`
- use `this.progressBar` in your modifier

I think that should work